models/IdSpecific.php

```diff
@@ -9,9 +9,10 @@ interface IdSpecific
 {
 
     /**
-     * Returns an entity.
+     * returns an entity.
      *
+     * @param int $id
      * @return object
      */
-    public function entity(): object;
+    public function entity(int $id): object;
```

Comment:
it was necessary to change the interface because it is not possible to store id inside the model itself.